pkg/controller/mpi_job_controller.go

```diff
@@ -585,9 +577,48 @@ func (c *MPIJobController) syncHandler(key string) error {
 		return err
 	}
 
+	if launcher != nil {
+		launcherSuspendUpdate := false
+		if isMPIJobSuspended(mpiJob) && !isJobSuspended(launcher) {
+			// suspend the launcher first if the MPI job is suspended
+			launcherSuspendUpdate = true
+			launcher.Spec.Suspend = pointer.Bool(true)
+		} else if !isMPIJobSuspended(mpiJob) && isJobSuspended(launcher) {
+			launcherSuspendUpdate = true
+			// unsuspend the launcher first if the MPI job is unsuspended
+			launcher.Spec.Suspend = pointer.Bool(false)
+		}
+		if launcherSuspendUpdate {
+			if _, err := c.kubeClient.BatchV1().Jobs(namespace).Update(context.TODO(), launcher, metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+
+	// cleanup the running worker pods if the MPI job is suspended
+	if isMPIJobSuspended(mpiJob) {
+		if err := cleanUpPods(mpiJob, c); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+func cleanUpPods(mpiJob *kubeflow.MPIJob, c *MPIJobController) error {
+	// set worker StatefulSet Replicas to 0.
+	if err := c.deleteWorkerPods(mpiJob); err != nil {
+		return err
+	}
+	initializeMPIJobStatuses(mpiJob, kubeflow.MPIReplicaTypeWorker)
+	if c.gangSchedulerName != "" {
+		if err := c.deletePodGroups(mpiJob); err != nil {
+			return err
+		}
+	}
```

Comment:
@tenzen-y are you familiar with the volcano integration?

I wonder if we need to remove the pod group on suspension. Does it matter?

Reply:
AFAIK, a podgroup is just a declaration that some pods should be treated as a unit. But a podgroup doesn't create other objects.

Reply:
The `scheduling.volcano.sh/v1beta1` PodGroups also have queueing logic. So we might to need to delete PodGroup to re-queue Pods.

Reply:
Maybe, we should create E2E with the volcano in a separate PR.

Reply:
probably someone with volcano experience should do it.

Reply:
I looked at the volcano code a little bit on my own, but could not conclude if it may create the pods under some scenarios.

Thus, for now, I delete the PodGroups either, to be on the safe side. WDYT?

+1 for the idea of an e2e test employing with podgroups. I guess we could create a follow-up Issue and ask for people willing to do it.

Reply:
> Thus, for now, I delete the PodGroups either, to be on the safe side. WDYT?

That sounds good to me.